libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -105,6 +105,36 @@ void CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
                 return;
             }
         }
+
+    } else if (("cpu.shares" == settingKey)) {
+        long int newValue;
```

Comment:
You may be right. There doesn't seem to be a limit, but I didn't know if we wanted to limit the user, but I do agree that an int should be sufficient.